spec/index.spec.js

```diff
@@ -531,6 +531,58 @@ describe('server', () => {
       .catch(done.fail);
   });
 
+  it('can create server with middleware', async () => {
+    const obj = {
+      middleware: function (req, res, next) {
+        next();
+      },
+    };
+    const spy = spyOn(obj, 'middleware').and.callThrough();
+    const parseServer = await new Promise((resolve, reject) => {
+      const server = ParseServer.ParseServer({
+        ...defaultConfiguration,
+        ...{
+          appId: 'anOtherTestApp',
+          masterKey: 'anOtherTestMasterKey',
+          serverURL: 'http://localhost:12667/parse',
+          serverStartComplete(error) {
+            if (error) {
+              reject(error);
+            } else {
+              resolve(server);
+            }
+          },
+          middleware: obj.middleware,
+        },
+      });
+    });
+    const app = express();
+    app.use('/parse', parseServer);
+    await new Promise(resolve => app.listen(12667, resolve));
```

Comment:
I think you need to close the server after the tests..

Can you refactor

```const httpServer = app.listen(12667, resolve);```

then add `httpServer.close()` to the end of the test

Reply:
done.